library/src/morpheus-master/sass/themes/_base.scss

```diff
@@ -1,49 +1,98 @@
+// @returns 11 CSS Custom Properties available in :root
+//      named colorname-0 through colorname-10
+//      The HSL you provide in the parameters will be in the
+//      middle of the range at colorname-5.
+// @returns 11 CSS Classes using colorname prefixed with sakai-colorize
+//      These classes set the background-color to colorname and
+//      set the color to a value which will always pass WCAG AA 
+//      guidelines. Color will be selected from the following,
+//      whichever first passes the contrast requirement.
+//            colorname-1
+//            colorname-9
+//            white
+//            black
+//
+// Example usage 
+//      @include defineColorHSL(--myschool-gray, 214, 3%, 55%);
+// Returns CSS Custom Properties on :root
 @mixin defineColorHSL($color, $hue, $saturation, $lightness){
-  #{$color}: unquote("hsl(#{$hue}, #{$saturation}, #{$lightness})");
-  #{$color}-h: #{$hue};
-  #{$color}-s: #{$saturation};
-  #{$color}-l: #{$lightness};
 
-  #{$color}-0: tint-color(hsl($hue, $saturation, $lightness), 10);
-  #{$color}-1: tint-color(hsl($hue, $saturation, $lightness), 8);
-  #{$color}-2: tint-color(hsl($hue, $saturation, $lightness), 6);
-  #{$color}-3: tint-color(hsl($hue, $saturation, $lightness), 4);
-  #{$color}-4: tint-color(hsl($hue, $saturation, $lightness), 2);
-  #{$color}-5: hsl($hue, $saturation, $lightness);
-  #{$color}-6: shade-color(hsl($hue, $saturation, $lightness), 2);
-  #{$color}-7: shade-color(hsl($hue, $saturation, $lightness), 4);
-  #{$color}-8: shade-color(hsl($hue, $saturation, $lightness), 6);
-  #{$color}-9: shade-color(hsl($hue, $saturation, $lightness), 8);
-  #{$color}-10: shade-color(hsl($hue, $saturation, $lightness), 10);
+  $j: 10;
+  $i: 7;
+  @while $i > 0 {
+    #{$color}--lighter-#{$i}: tint-color(hsl($hue, $saturation, $lightness), $j);
+    // These individual hue, saturation, and lightness values could be used at a later time
+    // to increase flexibility of automating complementary colors programmatically. Not in
+    // use at this time.
+    // #{$color}--lighter-#{$i}-h: hue(tint-color(hsl($hue, $saturation, $lightness), $j));
+    // #{$color}--lighter-#{$i}-s: saturation(tint-color(hsl($hue, $saturation, $lightness), $j));
+    // #{$color}--lighter-#{$i}-l: lightness(tint-color(hsl($hue, $saturation, $lightness), $j));
+
+    .sakai-colorize#{$color}--lighter-#{$i} { 
+      color: color-contrast( 
+        tint-color(hsl($hue, $saturation, $lightness), $j), 
+        shade-color(hsl($hue, $saturation, $lightness), 8), 
+        tint-color(hsl($hue, $saturation, $lightness), 8) 
+      ); 
+      background-color: tint-color(hsl($hue, $saturation, $lightness), $j); 
+    }
+    $j: $j - 1.5;
+    $i: $i - 1;
+  }
+
+  #{$color}: hsl($hue, $saturation, $lightness);
+  // #{$color}-h: hue(hsl($hue, $saturation, $lightness));
+  // #{$color}-s: saturation(hsl($hue, $saturation, $lightness));
+  // #{$color}-l: lightness(hsl($hue, $saturation, $lightness));
```

Comment:
Will these be used? Are they needed?

Reply:
I pulled these out for now until I figure out the rest of the code needed to make proper use of them